### DISCUSSION TITLE
Hash Funtion on Aptos

### DISCUSSION THREAD

**Discussion by Tranduy1dol** [aptos-labs/aptos-developer-discussions#350] [Questions]
### Describe your question in detail.

I'm coding a Layer 1 verifier contract for Layer 2 on Aptos Move language, and I need a hash function that returns a 32-byte value like SHA-256. Other hash functions on Aptos return a longer value. Please help me.

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

N/A

**Comment by 0xjunha** [comment]
You can get 32-byte hash values from the `std::hash` and `aptos_std::aptos_hash` modules

`std::hash` offers:
- [sha2_256 and sha3_256](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/hash.move)

`aptos_std::aptos_hash` offers:
- [keccak256](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/hash.move#L35)
- [blake2b_256](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/hash.move#L69)

**Reply by Tranduy1dol** [reply]
Is there any way to get a 16-byte result from SHA-256 on Aptos Move?

**Reply by 0xjunha** [reply]
SHA-256 returns 32-byte result. If you want to get 16-byte value out of the result, you can use:
[vector::trim](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/vector.move#L126C16-L126C20)
or
[vector::trim_reverse](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/vector.move#L136)

to extract 16 bytes from the 32-byte result
